### ISSUE TITLE
Fails to build with kotlin 1.9

### ISSUE THREAD

**Issue by vladraduvidican** [raamcosta/compose-destinations#475]
Hello!

I'm getting this error since upgrading to Kotlin 1.9:

```
org.jetbrains.kotlin.backend.common.BackendException: Backend Internal error: Exception during psi2ir
File being compiled: (138,21) in G:/_Projects/BumtziHaus/Android/app/src/main/java/com/bumtzihaus/app/ui/screen/menu/content/BumtziMenu.kt
The root cause org.jetbrains.kotlin.psi2ir.generators.ErrorExpressionException was thrown at: org.jetbrains.kotlin.psi2ir.generators.ErrorExpressionGenerator.generateErrorSimpleName(ErrorExpressionGenerator.kt:105)
null: KtNameReferenceExpression:
BrowseSearchListDestination
	at org.jetbrains.kotlin.backend.common.CodegenUtil.reportBackendException(CodegenUtil.kt:253)
	at org.jetbrains.kotlin.backend.common.CodegenUtil.reportBackendException$default(CodegenUtil.kt:237)
	at org.jetbrains.kotlin.psi2ir.generators.DeclarationGenerator.generateMemberDeclaration(DeclarationGenerator.kt:75)
	at org.jetbrains.kotlin.psi2ir.generators.ModuleGenerator.generateSingleFile(ModuleGenerator.kt:73)
	at org.jetbrains.kotlin.psi2ir.generators.ModuleGenerator.generateModuleFragment(ModuleGenerator.kt:53)
	at org.jetbrains.kotlin.psi2ir.Psi2IrTranslator.generateModuleFragment(Psi2IrTranslator.kt:97)
	at org.jetbrains.kotlin.backend.jvm.JvmIrCodegenFactory.convertToIr(JvmIrCodegenFactory.kt:255)
	at org.jetbrains.kotlin.backend.jvm.JvmIrCodegenFactory.convertToIr(JvmIrCodegenFactory.kt:59)
	at org.jetbrains.kotlin.codegen.KotlinCodegenFacade.compileCorrectFiles(KotlinCodegenFacade.java:41)
	at org.jetbrains.kotlin.kapt3.AbstractKapt3Extension.contextForStubGeneration(Kapt3Extension.kt:280)
	at org.jetbrains.kotlin.kapt3.AbstractKapt3Extension.analysisCompleted(Kapt3Extension.kt:174)
	at org.jetbrains.kotlin.kapt3.ClasspathBasedKapt3Extension.analysisCompleted(Kapt3Extension.kt:104)
	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM$analyzeFilesWithJavaIntegration$2.invoke(TopDownAnalyzerFacadeForJVM.kt:104)
	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration(TopDownAnalyzerFacadeForJVM.kt:114)
	at org.jetbrains.kotlin.cli.jvm.compiler.TopDownAnalyzerFacadeForJVM.analyzeFilesWithJavaIntegration$default(TopDownAnalyzerFacadeForJVM.kt:88)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:256)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler$analyze$1.invoke(KotlinToJVMBytecodeCompiler.kt:42)
	at org.jetbrains.kotlin.cli.common.messages.AnalyzerWithCompilerReport.analyzeAndReport(AnalyzerWithCompilerReport.kt:115)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.analyze(KotlinToJVMBytecodeCompiler.kt:247)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli(KotlinToJVMBytecodeCompiler.kt:87)
	at org.jetbrains.kotlin.cli.jvm.compiler.KotlinToJVMBytecodeCompiler.compileModules$cli$default(KotlinToJVMBytecodeCompiler.kt:47)
	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:168)
	at org.jetbrains.kotlin.cli.jvm.K2JVMCompiler.doExecute(K2JVMCompiler.kt:53)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:100)
	at org.jetbrains.kotlin.cli.common.CLICompiler.execImpl(CLICompiler.kt:46)
	at org.jetbrains.kotlin.cli.common.CLITool.exec(CLITool.kt:101)
	at org.jetbrains.kotlin.incremental.IncrementalJvmCompilerRunner.runCompiler(IncrementalJvmCompilerRunner.kt:460)
	at org.jetbrains.kotlin.incremental.IncrementalJvmCompilerRunner.runCompiler(IncrementalJvmCompilerRunner.kt:62)
	at org.jetbrains.kotlin.incremental.IncrementalCompilerRunner.doCompile(IncrementalCompilerRunner.kt:476)
	at org.jetbrains.kotlin.incremental.IncrementalCompilerRunner.compileImpl(IncrementalCompilerRunner.kt:399)
	at org.jetbrains.kotlin.incremental.IncrementalCompilerRunner.compileNonIncrementally(IncrementalCompilerRunner.kt:280)
	at org.jetbrains.kotlin.incremental.IncrementalCompilerRunner.compile(IncrementalCompilerRunner.kt:124)
	at org.jetbrains.kotlin.daemon.CompileServiceImplBase.execIncrementalCompiler(CompileServiceImpl.kt:636)
	at org.jetbrains.kotlin.daemon.CompileServiceImplBase.access$execIncrementalCompiler(CompileServiceImpl.kt:101)
	at org.jetbrains.kotlin.daemon.CompileServiceImpl.compile(CompileServiceImpl.kt:1598)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.rmi/sun.rmi.server.UnicastServerRef.dispatch(UnicastServerRef.java:360)
	at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:200)
	at java.rmi/sun.rmi.transport.Transport$1.run(Transport.java:197)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:712)
	at java.rmi/sun.rmi.transport.Transport.serviceCall(Transport.java:196)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport.handleMessages(TCPTransport.java:587)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run0(TCPTransport.java:828)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.lambda$run$0(TCPTransport.java:705)
	at java.base/java.security.AccessController.doPrivileged(AccessController.java:399)
	at java.rmi/sun.rmi.transport.tcp.TCPTransport$ConnectionHandler.run(TCPTransport.java:704)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: org.jetbrains.kotlin.psi2ir.generators.ErrorExpressionException: null: KtNameReferenceExpression:
BrowseSearchListDestination
	at org.jetbrains.kotlin.psi2ir.generators.ErrorExpressionGenerator.generateErrorSimpleName(ErrorExpressionGenerator.kt:105)
	at org.jetbrains.kotlin.psi2ir.generators.StatementGenerator.visitSimpleNameExpression(StatementGenerator.kt:375)
	at org.jetbrains.kotlin.psi2ir.generators.StatementGenerator.visitSimpleNameExpression(StatementGenerator.kt:54)
	at org.jetbrains.kotlin.psi.KtNameReferenceExpression.accept(KtNameReferenceExpression.kt:59)
	at org.jetbrains.kotlin.psi2ir.generators.StatementGenerator.generateStatement(StatementGenerator.kt:70)
	at org.jetbrains.kotlin.psi2ir.generators.StatementGenerator.generateExpression(StatementGenerator.kt:83)
	at org.jetbrains.kotlin.psi2ir.generators.ArgumentsGenerationUtilsKt$pregenerateCall$1.invoke(ArgumentsGenerationUtils.kt:508)
	at org.jetbrains.kotlin.psi2ir.generators.ArgumentsGenerationUtilsKt$pregenerateCall$1.invoke(ArgumentsGenerationUtils.kt:508)
	at org.jetbrains.kotlin.psi2ir.generators.ArgumentsGenerationUtilsKt.generateValueArgumentUsing(ArgumentsGenerationUtils.kt:371)
	at org.jetbrains.kotlin.psi2ir.generators.ArgumentsGenerationUtilsKt.pregenerateValueArgumentsUsing(ArgumentsGenerationUtils.kt:755)
	at org.jetbrains.kotlin.psi2ir.generators.ArgumentsGenerationUtilsKt.pregenerateCallUsing(ArgumentsGenerationUtils.kt:518)
	at org.jetbrains.kotlin.psi2ir.generators.ArgumentsGenerationUtilsKt.pregenerateCall(ArgumentsGenerationUtils.kt:508)
	at org.jetbrains.kotlin.psi2ir.generators.BodyGenerator.generateEnumConstructorCall(BodyGenerator.kt:362)
	at org.jetbrains.kotlin.psi2ir.generators.BodyGenerator.generateEnumConstructorCallOrSuperCall(BodyGenerator.kt:348)
	at org.jetbrains.kotlin.psi2ir.generators.BodyGenerator.generateEnumEntryInitializer(BodyGenerator.kt:336)
	at org.jetbrains.kotlin.psi2ir.generators.ClassGenerator.generateEnumEntry(ClassGenerator.kt:538)
	at org.jetbrains.kotlin.psi2ir.generators.DeclarationGenerator.generateEnumEntryDeclaration(DeclarationGenerator.kt:105)
	at org.jetbrains.kotlin.psi2ir.generators.DeclarationGenerator.generateClassMemberDeclaration(DeclarationGenerator.kt:99)
	at org.jetbrains.kotlin.psi2ir.generators.ClassGenerator.generateMembersDeclaredInClassBody(ClassGenerator.kt:503)
	at org.jetbrains.kotlin.psi2ir.generators.ClassGenerator.generateClass(ClassGenerator.kt:102)
	at org.jetbrains.kotlin.psi2ir.generators.ClassGenerator.generateClass$default(ClassGenerator.kt:64)
	at org.jetbrains.kotlin.psi2ir.generators.DeclarationGenerator.generateClassOrObjectDeclaration(DeclarationGenerator.kt:108)
	at org.jetbrains.kotlin.psi2ir.generators.DeclarationGenerator.generateMemberDeclaration(DeclarationGenerator.kt:53)
	... 49 more

```

**Comment by raamcosta** [comment]
Have you updated ksp version?

**Comment by raamcosta** [comment]
Try ksp 1.9.0-1.0.12

**Comment by tommyzat** [comment]
I have updated KSP, this error is with that version

**Comment by ammargitham** [comment]
@raamcosta I know that 1.9.51 is not officially released, but since it was [available in maven](https://search.maven.org/artifact/io.github.raamcosta.compose-destinations/core/1.9.51/aar) and it seems to be build with Kotlin 1.9.0, I tried it out. But I still get the same error.

**Comment by raamcosta** [comment]
I actually forgot the release on GitHub 😅

but it’s really weird that you’re still getting the same error 🤔

**Comment by tommyzat** [comment]
I'm still getting the errors as well

**Comment by ammargitham** [comment]
Setting `kapt.use.jvm.ir=false` in `gradle.properties` seems to fix the error. But I'm not sure if that's the correct solution.

**Comment by tommyzat** [comment]
> Setting `kapt.use.jvm.ir=false` in `gradle.properties` seems to fix the error. But I'm not sure if that's the correct solution.
> 
> @raamcosta Any idea if this is the correct workaround/solution?

have you managed to find a solution?

**Comment by ammargitham** [comment]
> have you managed to find a solution?

I still have that property enabled. So far I have faced no issues.

**Comment by tommyzat** [comment]
I have found what causes this issue.

Enabling Hilt does this. Without it, it compiles fine.

**Comment by ammargitham** [comment]
Interesting. Hilt is still using kapt, so it makes sense that disabling a kapt property fixes the problem.

Fortunately, Hilt has released 1.1.0-alpha01 preparing for ksp support.

**Comment by raamcosta** [comment]
But maybe hilt team should know that it seems like some incompatibility with ksp and Kotlin 1.9. Because if that’s indeed the case they may want to do some fix for kapt version.
Had anyone checked if there’s an open issue on hilt side about this?

**Comment by rvenky125** [comment]
I'm also getting the same issue and this one `kapt.use.jvm.ir=false` not worked for me.
Please let me know if there are any solutions

**Comment by rvenky125** [comment]
The hilt 1.1.0-alpha01 worked.

**Comment by rvenky125** [comment]
It worked for the build but while opening the app getting this. Did any one encountered. Sorry for posting this here instead of at hilt's repo.

 FATAL EXCEPTION: main
                                                                                                    Process: com.example, PID: 23343
                                                                                                    java.lang.RuntimeException: Unable to instantiate application com.example.Application: java.lang.ClassNotFoundException: Didn't find class "com.example.Application" on path: DexPathList[[zip file "/data/app/com.exmaple-2/base.apk"],nativeLibraryDirectories=[/data/app/com.example-2/lib/x86, /data/app/com.example-2/base.apk!/lib/x86, /system/lib, /vendor/lib]]
                                                                                                    	at android.app.LoadedApk.makeApplication(LoadedApk.java:802)
                                                                                                    	at android.app.ActivityThread.handleBindApplication(ActivityThread.java:5335)
                                                                                                    	at android.app.ActivityThread.-wrap2(ActivityThread.java)
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1528)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:102)
                                                                                                    	at android.os.Looper.loop(Looper.java:154)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:6077)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:866)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:756)
                                                                                                    Caused by: java.lang.ClassNotFoundException: Didn't find class "com.example.Application" on path: DexPathList[[zip file "/data/app/com.example-2/base.apk"],nativeLibraryDirectories=[/data/app/com.example-2/lib/x86, /data/app/com.example-2/base.apk!/lib/x86, /system/lib, /vendor/lib]]

**Comment by rvenky125** [comment]
> Interesting. Hilt is still using kapt, so it makes sense that disabling a kapt property fixes the problem.
> 
> Fortunately, Hilt has released 1.1.0-alpha01 preparing for ksp support.

Did it work for you?
